src/pages/instances/actions/AttachIsoBtn.tsx

```diff
@@ -0,0 +1,120 @@
+import { useParams } from "react-router-dom";
+import { useNotify } from "@canonical/react-components";
+import { getInstanceEditValues, getInstancePayload } from "util/instanceEdit";
+import { LxdIsoDevice } from "types/device";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import usePortal from "react-useportal";
+import { RemoteImage } from "types/image";
+import SubmitButton from "components/SubmitButton";
+import CustomIsoModal from "pages/images/CustomIsoModal";
+
+interface Props {
+  instance: LxdInstance;
+}
+
+const AttachIsoBtn: FC<Props> = ({ instance }) => {
+  const { project } = useParams<{ project: string }>();
+  const notify = useNotify();
+  const queryClient = useQueryClient();
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  const [isLoading, setLoading] = useState(false);
+
+  const attachedIso = instance.devices["iso-volume"] as
+    | LxdIsoDevice
+    | undefined;
+
+  const ejectIso = () => {
+    setLoading(true);
+    const values = getInstanceEditValues(instance);
+    values.devices = values.devices.filter((device) => {
+      return device.name !== "iso-volume";
+    });
+    const instanceMinusIso = getInstancePayload(
+      instance,
+      values
+    ) as LxdInstance;
+    updateInstance(instanceMinusIso, project ?? "")
+      .then(() =>
+        notify.success(
+          <>
+            ISO <b>{attachedIso?.source ?? ""}</b> ejected
+          </>
+        )
+      )
+      .catch((e: Error) => notify.failure("Detach ISO failed", e))
+      .finally(() => {
+        void queryClient.invalidateQueries({
+          queryKey: [queryKeys.instances, instance.name, project],
+        });
+        setLoading(false);
+      });
+  };
+
+  const handleSelect = (image: RemoteImage) => {
+    setLoading(true);
+    closePortal();
+    const values = getInstanceEditValues(instance);
+    values.devices.push({
+      type: "iso-volume",
+      name: "iso-volume",
+      bare: {
+        "boot.priority": "10",
+        pool: image.pool ?? "",
+        source: image.aliases,
+        type: "disk",
+      },
+    });
+    const instancePlusIso = getInstancePayload(instance, values) as LxdInstance;
+    updateInstance(instancePlusIso, project ?? "")
+      .then(() =>
+        notify.success(
+          <>
+            ISO <b>{image.aliases}</b> attached
+          </>
+        )
+      )
+      .catch((e: Error) => notify.failure("Attaching ISO failed", e))
+      .finally(() => {
+        void queryClient.invalidateQueries({
+          queryKey: [queryKeys.instances, instance.name, project],
+        });
+        setLoading(false);
+      });
+  };
+
+  return attachedIso ? (
+    <>
+      <span className="u-text--muted margin-right">{attachedIso.source}</span>
+      <SubmitButton
+        appearance=""
+        isSubmitting={isLoading}
+        isDisabled={false}
+        buttonLabel="Detach ISO"
```

Comment:
Here too, I suggest using the verb "eject", as it is probably more familiar in this kind of context, but if there's a particular reason you chose "detach", feel free to disregard this suggestion.
```suggestion
        buttonLabel="Eject ISO"
```